File: labs/lab02/exercise1.py
##
# This program calculate the number of used tiles, gap on each side and number of used black and white tiles.
#

totalWidth = 23
print("Total width =", totalWidth)

tileWidth = 2
print("Tile width =", tileWidth)

# Write your code below this line
ADDITIONAL_BLACK_TILE = 1
sequenceLength = 2

# Calculate the of number of pairs in the whole sequence after the first black tile
widthWithoutBlackTile = totalWidth - tileWidth
numberOfWhilteBlackTilePairs = widthWithoutBlackTile // (sequenceLength * tileWidth)

# Calculate the number of tiles used in whole sequence
numberOfTiles = sequenceLength * numberOfWhilteBlackTilePairs + ADDITIONAL_BLACK_TILE
print("Number of tiles =", numberOfTiles)

# Calculate the length of gap on each side
lengthOfUsedTiles = numberOfTiles * tileWidth
totalGap = totalWidth - lengthOfUsedTiles
gapِAtEachEnd = totalGap / 2
print("Gapِ at each end =", gapِAtEachEnd)

# Calculate the number of black tiles with the first tile
numberOfBlackTiles = numberOfWhilteBlackTilePairs + ADDITIONAL_BLACK_TILE
print("Number of Black Tiles", numberOfBlackTiles)

# Calculate the number of white tiles
numberOfWhiteTiles = numberOfWhilteBlackTilePairs
print("Number of White Tiles", numberOfWhiteTiles)

File: labs/lab12/exercise1.c
#include <stdio.h>
#define ADDITIONAL_BLACK_TILE 1

int main(void) {
    int totalWidth, tileWidth, widthWithOutBlackTile,
        numberOfWhiteBlackTilePairs, numberOfTiles,
        numberOfBlackTiles, numberOfWhiteTiles,
        lengthOfTilesUsed, totalGap;

    double gapOnEashSide;

    totalWidth = 23;
    tileWidth = 2;

    widthWithOutBlackTile = totalWidth - tileWidth;
    numberOfWhiteBlackTilePairs = widthWithOutBlackTile/(2 * tileWidth);
    numberOfTiles = 1 + 2 * numberOfWhiteBlackTilePairs;
    lengthOfTilesUsed = numberOfTiles * tileWidth;
    totalGap = totalWidth - lengthOfTilesUsed;
    gapOnEashSide = totalGap / 2.0;

    numberOfBlackTiles = numberOfWhiteBlackTilePairs + ADDITIONAL_BLACK_TILE;
    numberOfWhiteTiles = numberOfWhiteBlackTilePairs;

    printf("Total Width = %d\n", totalWidth);
    printf("Tile width = %d\n", tileWidth);
    printf("Number of tiles %d\n", numberOfTiles);
    printf("Gap on each sides = %.1lf\n", gapOnEashSide);
    printf("Number of black tiles %d\n", numberOfBlackTiles);
    printf("Number of white tiles %d\n", numberOfWhiteTiles);
}